### PR TITLE
Better api key update

### PR DESCRIPTION
**Pull request by lucasmccabe**
This is a small PR that accomplishes two goals:
- We check if check if an OpenAI key provided via the UI is of the correct/expected format before proceeding to make queries.
- We conceal an OpenAI key provided via the UI as it's being typed.

Illustrated below:

image

### CODE REVIEW

File: src/Generation.jl [hunk 1]
```diff
@@ -17,13 +17,43 @@ export OAIGenerator,
...
+- https://github.com/secretlint/secretlint/issues/676
+- https://community.openai.com/t/what-are-the-valid-characters-for-the-apikey/288643
+
+Note that this only checks the key format, not whether the key is valid or has not 
+been revoked.
+"""
+function check_oai_key_format(key::String)
+    pattern = r"^sk-[A-Za-z0-9]{20}T3BlbkFJ[A-Za-z0-9]{20}$"
+    result = occursin(pattern, key)
+    return result
```
Comment (toon-leader-bacon) [inline src/Generation.jl:54]:
```suggestion
    return occursin(pattern, key)
```

Minor suggestion, not critical.

File: src/index.html [hunk 2]
```diff
@@ -45,7 +45,7 @@ <h3>Corpus Manager</h3>
 
             <!-- content for the API Key tab (just for storing/updating the key) -->
             <div id = "api-key" class = "tab-content" style = "display: none;">
-                <input type = "text" id = "api-key-field" placeholder = "OpenAI API Key">
+                <input type = "password" id = "api-key-field" placeholder = "OpenAI API Key">
```
Comment (toon-leader-bacon) [inline src/index.html:48]: This is cool. I take it this is the thing that makes the user typed text into `****`? I guess I learn something new every day.

Comment (lucasmccabe) [reply src/index.html:48]: Yes! TIL for me, as well.

File: src/Generation.jl (hunk 1 above)
Comment (lucasmccabe) [reply src/Generation.jl:54]: Good one, this was actually a debugging artifact.

**Comment by toon-leader-bacon** [comment]
Hmm, I'm playing around with running the `Frontend.jl` file. I can load my openAI API key, and can load the two provided test corpus, but I'm unable to actually query anything from it. I just get the error popup. 
image


Any thoughts?

**Comment by lucasmccabe** [comment]
@toon-leader-bacon After selecting the corpus from the drop-down menu, did you press the load button? If successful, you should see a pop-up like this:

image

**Comment by toon-leader-bacon** [comment]
Yep! 
image

Whatever, I don't think it's super critical right now. I think it's more a me problem than a proposed code problem. I'll approve and try to play around with things on my end to see if anything works. But getting more documentation for this would be quite helpful. Maybe as I'm debugging/ playing I'll try to take notes and write up some simple documentation for Alex to use in his.

**Review by toon-leader-bacon — APPROVED** [review]

**Pull request merged**